src/mkdocstrings/handlers/python.py

```diff
@@ -290,3 +294,30 @@ def rebuild_category_lists(obj: dict) -> None:
     obj["children"] = [child for _, child in obj["children"].items()]
     for child in obj["children"]:
         rebuild_category_lists(child)
+
+
+def sort_object(obj: CollectorItem, sort_style: str) -> None:
+    """Sort the collected object's children.
+
+    Sorts the object's children list, then each category separately, and then recurses into each.
+
+    Arguments:
+        obj: The collected object, as a dict. Note that this argument is mutated.
+        sort_style: How to sort the children lists - 'lexical' or 'source'.
+    """
+
+    def sort_key(a: dict) -> Any:
+        if sort_style == "lexical":
+            return a.get("name")
+        elif sort_style == "source":
+            return a.get("source", {}).get("line_start", 0)
+        else:
+            raise Exception("unknown sort_style")
+
+    obj["children"].sort(key=sort_key)
+
+    for category in ("attributes", "classes", "functions", "methods", "modules"):
+        obj[category].sort(key=sort_key)
+
+    for child in obj["children"]:
+        sort_object(child, sort_style=sort_style)
```

Comment:
There's probably some more specific way to throw errors. Maybe `PluginError`?

In the future we could consider validating configs at the very beginning, but I think the infrastructure is currently not there, so disregard that part.

Reply:
`PluginError` makes sense to me!